src/shogun/regression/LinearRidgeRegression.cpp

```diff
@@ -32,57 +32,64 @@ CLinearRidgeRegression::CLinearRidgeRegression(float64_t tau, CDenseFeatures<flo
 void CLinearRidgeRegression::init()
 {
 	set_tau(1e-6);
+	m_use_bias = true;
 
 	SG_ADD(&m_tau, "tau", "Regularization parameter", MS_AVAILABLE);
+	SG_ADD(
+	    &m_use_bias, "use_bias", "Whether or not to fit an offset term",
+	    MS_NOT_AVAILABLE);
 }
 
-bool CLinearRidgeRegression::train_machine(CFeatures* data)
+template <typename T>
+bool CLinearRidgeRegression::train_machine_templated(
+    const CDenseFeatures<T>* feats)
 {
-    REQUIRE(m_labels,"No labels set\n");
-
-    if (!data)
```

Comment:
thanks to @shubham808 s work, we can now delete such redundant codes, while making algorithms fully templated